statistic/src/main/java/org/polypheny/db/statistic/exploreByExample/Explore.java

```diff
@@ -0,0 +1,277 @@
+                    }
+                } else if ( dataType[dim].equals( "INTEGER" ) || dataType[dim].equals( "BIGINT" ) ) {
+                    vals[dim] = Double.parseDouble( table.get( obj )[dim] );
+                }
+            }
+            classifiedData.add( new DenseInstance( 1.0, vals ) );
+        }
+        return classifiedData;
+    }
+
+    public J48 trainData( Instances classifiedData){
+        classifiedData.setClassIndex( classifiedData.numAttributes() - 1 );
+        J48 tree = new J48();
+
+        String[] options = { "-U" };
+        try {
+            tree.setOptions( options );
+        } catch ( Exception e ) {
+            log.error("setting options for Classification not possible");
+            e.printStackTrace();
+        }
+
+        try {
+            tree.buildClassifier( classifiedData );
+        } catch ( Exception e ) {
+            log.error("build Classifier for tree not possible");
+            e.printStackTrace();
+        }
+
+        try {
+            this.buildGraph = tree.graph();
+        } catch ( Exception e ) {
+            log.error("build graph tree not possible");
+            e.printStackTrace();
+        }
+
+        return tree;
+    }
+
+    public List<String[]> classifyUnlabledData( J48 tree, Instances unlabeled ){
+
+        unlabeled.setClassIndex( unlabeled.numAttributes() - 1 );
+        Instances labeled = new Instances( unlabeled );
+        String[] label = new String[unlabeled.numInstances()];
+        List<String[]> labledData = new ArrayList<>();
+
+        for ( int i = 0; i < unlabeled.numInstances(); i++ ) {
+            double clsLabel = 0;
+
+            try {
+                clsLabel = tree.classifyInstance( unlabeled.instance( i ) );
+            } catch ( Exception e ) {
+                log.error("classification of unlabeled data not possible");
+                e.printStackTrace();
+            }
+
+            labeled.instance( i ).setClassValue( clsLabel );
+            label[i] = unlabeled.classAttribute().value( (int) clsLabel );
+            labledData.add( labeled.instance( i ).toString().split( "," ) );
+        }
+
+        return labledData;
+    }
+
+
+    /**
+     * Classify all Data with tree built before
+     * @param tree J48 Weka Tree
+     * @param unlabeled all selected unlabeled Data
+     * @return only the data labeled as true
+     */
+    public String[][] classifyData( J48 tree, Instances unlabeled ){
+        List<String[]> labledData = new ArrayList<>();
+        unlabeled.setClassIndex( unlabeled.numAttributes() - 1 );
+        Instances labeled = new Instances( unlabeled );
+
+        for ( int i = 0; i < unlabeled.numInstances(); i++ ) {
+            double clsLabel = 0;
+
+            try {
+                clsLabel = tree.classifyInstance( unlabeled.instance( i ) );
+            } catch ( Exception e ) {
+                log.error("classification of unlabeled data not possible");
```

Comment:
```suggestion
                log.error("classification of unlabeled data not possible", e);
```

Reply:
resolved in commit:[https://github.com/polypheny/Polypheny-DB/pull/211/commits/5aa89f0c4a4106cfc45536101cf1877310f5dea9](url)